backend/Infrastructure/Repositories/Staffings/StaffingCacheRepository.cs

```diff
@@ -0,0 +1,78 @@
+using Core.Staffings;
+using Microsoft.Extensions.Caching.Memory;
+
+namespace Infrastructure.Repositories.Staffings;
+
+public class StaffingCacheRepository(IStaffingRepository sourceRepository, IMemoryCache cache) : IStaffingRepository
+{
+    public async Task<Dictionary<int, List<Staffing>>> GetStaffingForConsultants(List<int> consultantIds,
+        CancellationToken ct)
+    {
+        var nonCachedIds = new List<int>();
+        var result = new Dictionary<int, List<Staffing>>();
+
+        foreach (var consultantId in consultantIds)
```

Comment:
```suggestion
        foreach (var consultantId in consultantIds.Distinct())
```

Kan redde en footgun her